lib/app/server.js

```diff
@@ -81,7 +81,7 @@ export default context => {
     // nuxtServerInit
     <% if (store) { %>
       let promise = (store._actions && store._actions.nuxtServerInit ? store.dispatch('nuxtServerInit', omit(getContext(context), 'redirect', 'error')) : null)
-      if (!(promise instanceof Promise)) promise = Promise.resolve()
+      if (!promise || (!(promise instanceof Promise) && (promise && typeof promise.then !== 'function'))) promise = Promise.resolve()
```

Comment:
@Atinux 
axios or some other frameworks return  promise-like object, NOT a ES's Promise object. 
So the statement `promise instanceof Promise` will return `false`.
But a promise-like object behave like a promise and do works with ES's Promise object. 
Simply, just use Promise.resolve(xxx) will compatible with both promise and promise-like object
> ` promise = Promise.resolve(promise);`

BTW, to check if a object is a promise-like or not, `typeof promise.then !== 'function'`  is not enough.